code/modules/surgery/robotics.dm

```diff
@@ -666,3 +666,44 @@
 	user.visible_message("<span class='warning'> [user]'s [tool.name] slips, failing to reprogram [target]'s [affected.name].</span>",
 	"<span class='warning'> Your [tool.name] slips, failing to reprogram [target]'s [affected.name].</span>")
 	return SURGERY_STEP_RETRY
+
+/datum/surgery/robotics/reconfigure_id
+	name = "Reconfigure Identity"
+	steps = list(
+		/datum/surgery_step/robotics/external/unscrew_hatch,
+		/datum/surgery_step/robotics/external/open_hatch,
+		/datum/surgery_step/robotics/edit_serial,
+		/datum/surgery_step/robotics/external/close_hatch)
+	possible_locs = list(BODY_ZONE_CHEST, BODY_ZONE_HEAD)
+
+/datum/surgery_step/robotics/edit_serial
+	name = "edit serial number"
+	allowed_tools = list(TOOL_MULTITOOL = 100)
+	time = 4.8 SECONDS
+
+/datum/surgery_step/robotics/edit_serial/begin_step(mob/living/user, mob/living/carbon/human/target, target_zone, obj/item/tool,datum/surgery/surgery)
+	user.visible_message(
+		"[user] begins to edit [target]'s identity parameters with [tool].",
+		"You begin to alter [target]'s identity parameters with [tool]."
+		)
+	return ..()
+
+/datum/surgery_step/robotics/edit_serial/end_step(mob/living/user, mob/living/carbon/human/target, target_zone, obj/item/tool,datum/surgery/surgery)
+	var/old_name = target.real_name
+	var/new_name = copytext(reject_bad_text(input(user,"Choose a name for this machine.","Set Name","[old_name]") as null|text),1,MAX_NAME_LEN)
+	if(!new_name)
+		to_chat(user, "<span class='warning'> Invalid name! Please try again. </span>")
+		return SURGERY_STEP_RETRY
+	else
```

Comment:
You can also make these early returns. Instead of the if/else, just remove the else and unindent everything below here.

You can do the same after the `new_gender` nullcheck as well.